### ISSUE TITLE
Direct Access to Node-RED Device Editor

### ISSUE THREAD

**Issue by MarianRaphael** [FlowFuse/flowforge#2855]
### Description

Add an option in the Device Settings to allow direct access to the Node-RED Device Editor. The feature should include authentication that syncs with FlowFuse, similar to how the NR-Tools-Plugin works.

## Details:
Currently, users have to navigate multiple steps to access the Node-RED Device Editor. This can be cumbersome and time-consuming, particularly when adjustments are needed frequently.

## Proposed Solution:
- Integrate a "Direct Access to Node-RED Editor" option within the Settings of each Device.
- Implement an authentication process that aligns with the existing FlowFuse system. This could be modeled after the NR-Tools-Plugin's authentication mechanism.

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

**Comment by MarianRaphael** [comment]
@knolleary, as an alternative approach to direct access, would it be possible to simply provide a 'Device Editor URL'? When a user clicks on this link, the following actions would automatically occur:

Developer Mode Enabled (via API)
Device Editor Enabled (via API)
Open the Device Editor

From the user's perspective, it would appear to be the same as direct access.
(see also: https://github.com/FlowFuse/flowfuse/issues/2884)

**Comment by knolleary** [comment]
One of the reasons to enabling direct access is to remove the need to tunnel all traffic through the platform when the user is sat physically next to the device and on the same network.

What is proposed in #2884 improves the UX for remote access, but I believe there is still value allowing direct access.

**Comment by ZJvandeWeg** [comment]
@knolleary @MarianRaphael I don't believe we should enable direct access. It circumvents a key value pillar of FlowFuse: Compliance and Security.

The editor being available from FlowFuse, https://github.com/FlowFuse/flowfuse/issues/2246, seems to be a better alternative.

**Comment by knolleary** [comment]
Direct access would still be done securely and require an authentication against the FF platform - with auditing etc in place. I don't think that is a reason to discount this idea.

**Comment by ZJvandeWeg** [comment]
@knolleary Let's have one way to do it, not 2 or more: https://flowfuse.com/handbook/product/strategy/#convention-over-configuration
